### PR TITLE
Get comments by article

### PR DESCRIPTION
**Pull request by mohsinahmxd**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -154,3 +154,63 @@ describe('testing GET /api/articles', () => {
...
+        .expect(404)
+        .then((data) => {
+            expect(data._body.msg).toBe("No article found for article_id: 999999999");
+        })
+    });
+    test('should return error 400 and a message when passed an invalid id', () => {
+        return request(app).get("/api/articles/notAnId/comments")
+        .expect(400)
+        .then(data => {
+            expect(data._body.msg).toBe("Invalid input: 400 Bad Request");
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:201]: Grammatically speaking: '400 Bad Request: Invalid input' makes more sense :)

File: controllers/app.controller.js [hunk 2]
```diff
@@ -30,5 +30,16 @@ function getAllArticles(request, response, next) {
     })
 }
 
-module.exports = {topicsController, apiController, articleController, getAllArticles}
+function getAllCommentsForArticle(request, response, next) {
+    const chosenArticleId = request.params.article_id
+
+    return getAllCommentsForArticleModel(chosenArticleId).then(data => {
+        console.log(data);
```
Comment (nc-niamh-smith) [inline controllers/app.controller.js:37]: please remove console.logs :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -154,3 +154,63 @@ describe('testing GET /api/articles', () => {
...
+        .then(data => {
+            let resultArr = data._body.comments; // putting in variable for convenience
+            expect(resultArr.length > 0).toEqual(true); // check for empty arr
+            for (let i = 0; i < resultArr.length - 2; i++) { // loop until 2nd last
+                // convert strings to date then to numbers to compare
+                expect(new Date(resultArr[i].created_at).getTime()).toBeGreaterThanOrEqual(new Date(resultArr[i + 1].created_at).getTime())
+            }
+        })
+    });
+});
```
Comment (nc-niamh-smith) [inline __tests__/app.test.js:216]: Just missing a test for if the id is valid but there are no comments (it'll be a 200, and the response should be an empty array!) - this will impact your model!!

**Review by nc-niamh-smith — COMMENTED** [review]
Lovely work here Mohsin! A test to add and just a couple of notes for you :)
